### ISSUE TITLE
Can't use volumes with docker-compose

### ISSUE THREAD

**Issue by stereobooster** [agentejo/cockpit-docker#6]
docker-compose.yml

```yml
version: "3"

services:
  web:
    image: agentejo/cockpit
    ports:
      - "8080:80"
    restart: always
    volumes:
      - ./db/storage/:/var/www/html/storage
      - ./db/config/:/var/www/html/config
```

Run

```sh
docker-compose up
```

Open http://localhost:8080/install/ and get

```
Data folder is not writable: /storage/data
Cache folder is not writable: /storage/cache
Temp folder is not writable: /storage/tmp
Uploads folder is not writable: /storage/uploads
```

https://github.com/stereobooster/headless-cms-comparison/tree/master/cockpit

**Comment by ufoscout** [comment]
Any update on this?
My company is evaluating cockpit and this is a blocker for us.

**Comment by aheinze** [comment]
my solution to this is to create a custom `container-entrypoint.sh` with the following contents:

```
#!/bin/sh

mkdir -p /var/www/html/storage/data
mkdir -p /var/www/html/storage/uploads
mkdir -p /var/www/html/storage/cache
mkdir -p /var/www/html/storage/thumbs
mkdir -p /var/www/html/storage/tmp

chown -R www-data:www-data /var/www/html/

exec $@
```

and then in the docker file use the file as the entry point:

```
ADD container-entrypoint.sh /
ENTRYPOINT ["/container-entrypoint.sh"]
```

**Comment by hilyin** [comment]
I am stuck on this also. What is weird, is locally on OSX docker, it works fine. But on ubuntu 18 docker, I get this issue:

![image](https://user-images.githubusercontent.com/6288867/68832332-84575180-0665-11ea-94c6-e73643e2e614.png)

**Comment by tommueller** [comment]
I also ran into this issue. Could the suggestions proposed by @aheinze not be added to the official dockerfile?

**Comment by cupcakearmy** [comment]
fixed it in this pr https://github.com/agentejo/cockpit-docker/pull/12

**Comment by cupcakearmy** [comment]
@stereobooster test it now with the pr merged. If so you can close the issue :)

**Comment by tommueller** [comment]
Fixed it for me. Thx @cupcakearmy && @aheinze !

**Comment by BlindDespair** [comment]
@cupcakearmy still not working for me. Getiing this error:
![Screenshot from 2020-01-17 00-33-45](https://user-images.githubusercontent.com/15785773/72571912-3bb85080-38c1-11ea-8425-b82659b328df.png)
![Screenshot from 2020-01-17 00-35-40](https://user-images.githubusercontent.com/15785773/72571949-5985b580-38c1-11ea-99d3-a3158f2e1f9a.png)

I tried just using docker-compose.yml file from the original Issue message. Also I tried clonning repository and building image locally, the issue stays the same. Any idea?

**Comment by cupcakearmy** [comment]
Is it a clean system or an existing one?

**Comment by cupcakearmy** [comment]
@BlindDespair have you tried these steps?

https://github.com/agentejo/cockpit-docker/pull/12#issue-358825446

**Comment by BlindDespair** [comment]
@cupcakearmy both me and my friend tried it. My system was not completely clean, because I already had cockpit running before although it was just couple of weeks. But my friend yesterday did it on completely fresh system and had the same issue.

I tried the steps you mentioned there. I checked out the repository, built a docker image with docker-compose and then tried running but still same problem...

**Comment by cupcakearmy** [comment]
if 3.7 is an invalid version you should update docker & docker-compose first. maybe there is the error?

**Comment by cupcakearmy** [comment]
i don't know what else could be the issue

**Comment by BlindDespair** [comment]
@cupcakearmy Thank you for fast responses. I will check my version today in the evening. But I installed docker and docker-compose recently with apt-get on ubuntu, I think it should be already one of the most recent. But if it turns out to be the case, I think still won't completely be okay if I can't just install docker by default and use it that way.
Anyway I will let you know if I find out anything.

**Comment by cupcakearmy** [comment]
Have you added the docker repository or have you just `apt install docker` because that installs a super old verison. you need `docker-ce`

**Comment by cupcakearmy** [comment]
If you haven't follow these instructions https://docs.docker.com/install/linux/docker-ce/ubuntu/

**Comment by BlindDespair** [comment]
@cupcakearmy thank you again for your answers very much. I don't remember now how exactly I installed I think I had to add something. But I will check in the evening again, it could be that I did it wrong actually.

**Comment by BlindDespair** [comment]
@cupcakearmy I can confirm that it works when I checkout the project and build image myself. However when I run it using proper docker-compose.yml file using image from registry it seems like entrypoint is just not running for me or not doing anything.
Here is my docker-compose.yml:
```
version: "3.7"

services:
  web:
    image: agentejo/cockpit
    ports:
      - "8080:80"
    volumes:
      - ./db/storage/:/var/www/html/storage
      - ./db/config/:/var/www/html/config
```
when I do `ls -l` on db folder I see root as the only owner on it and on the folders inside it too. However when I check permissions in checked out cockpit-docker with locally build image I have only root as owner on db folder but I have www-data as owner on config and storage which is why it works.
Now what I don't get is why does it behave differently?

**Comment by BlindDespair** [comment]
Ok, I figured out finally what happened here and why it was different. Apparently I had cockpit pulled before your fix. And somehow even despite me writing `:latest` in some tries it would always use the image that was already pulled, so basically it was just cached. I removed the image locally and then triggered another build, it pulled new image and it finally works! I can confirm that it's working.
